pkg/ccl/logictestccl/testdata/logic_test/partitioning

```diff
@@ -1411,3 +1411,89 @@ vectorized: true
+
+statement ok
+CREATE TABLE bar(a int primary key) PARTITION BY LIST (a) (PARTITION x VALUES IN (1), PARTITION y VALUES IN (2));
+
+statement ok
+ALTER PARTITION x OF TABLE bar CONFIGURE ZONE USING gc.ttlseconds=100;
+
+statement ok
+ALTER PARTITION y OF TABLE bar CONFIGURE ZONE USING gc.ttlseconds=150;
+
+query TT
+SHOW CREATE TABLE bar;
+----
+bar  CREATE TABLE public.bar (
+       a INT8 NOT NULL,
+       CONSTRAINT bar_pkey PRIMARY KEY (a ASC)
+     ) PARTITION BY LIST (a) (
+       PARTITION x VALUES IN ((1)),
+       PARTITION y VALUES IN ((2))
+     );
+     ALTER PARTITION x OF INDEX test.public.bar@bar_pkey CONFIGURE ZONE USING
+       gc.ttlseconds = 100;
+     ALTER PARTITION y OF INDEX test.public.bar@bar_pkey CONFIGURE ZONE USING
+       gc.ttlseconds = 150
+
+
+statement ok
+TRUNCATE bar;
+
+query TT
+SHOW CREATE TABLE bar;
+----
+bar  CREATE TABLE public.bar (
+       a INT8 NOT NULL,
+       CONSTRAINT bar_pkey PRIMARY KEY (a ASC)
+     ) PARTITION BY LIST (a) (
+       PARTITION x VALUES IN ((1)),
+       PARTITION y VALUES IN ((2))
+     );
+     ALTER PARTITION x OF INDEX test.public.bar@bar_pkey CONFIGURE ZONE USING
+       gc.ttlseconds = 100;
+     ALTER PARTITION y OF INDEX test.public.bar@bar_pkey CONFIGURE ZONE USING
+       gc.ttlseconds = 150
+
+query TT
+SHOW ZONE CONFIGURATION FOR PARTITION x OF INDEX test.public.bar@bar_pkey;
+----
+PARTITION x OF INDEX test.public.bar@bar_pkey  ALTER PARTITION x OF INDEX test.public.bar@bar_pkey CONFIGURE ZONE USING
+                                                 range_min_bytes = 134217728,
+                                                 range_max_bytes = 536870912,
+                                                 gc.ttlseconds = 100,
+                                                 num_replicas = 3,
+                                                 constraints = '[]',
+                                                 lease_preferences = '[]'
+
+query TT
+SHOW ZONE CONFIGURATION FOR PARTITION y OF INDEX test.public.bar@bar_pkey;
+----
+PARTITION y OF INDEX test.public.bar@bar_pkey  ALTER PARTITION y OF INDEX test.public.bar@bar_pkey CONFIGURE ZONE USING
+                                                 range_min_bytes = 134217728,
+                                                 range_max_bytes = 536870912,
+                                                 gc.ttlseconds = 150,
+                                                 num_replicas = 3,
+                                                 constraints = '[]',
+                                                 lease_preferences = '[]'
+
+statement ok
+CREATE VIEW subzone_spans AS SELECT json_array_elements(crdb_internal.pb_to_json('cockroach.config.zonepb.ZoneConfig', config)->'subzoneSpans')->> 'key' AS key FROM system.zones WHERE id = 'bar'::REGCLASS::OID;
+
+query T rowsort
+SELECT * FROM subzone_spans;
+----
+iok=
+ioo=
+
+query T rowsort
+SELECT crdb_internal.pretty_key(decode(key, 'base64'), -1) FROM subzone_spans;
```

Comment:
I added this test to illustrate that our subzone spans are rewritten for the new index id (2)